benchmark/kvbench/runtime/etcd_rt.py

```
@@ -168,70 +207,112 @@ def allgather_obj(self, obj: Any) -> List[Any]:
-            val = self.client.get(f"{self.prefix}/alltoall/{src_rank}_to_{self.rank}")[
-                0
-            ]
+            key = f"{self.prefix}/alltoall/{alltoall_ix}/{src_rank}_to_{self.rank}"
+            val = self.client.get(key)[0]
+            # Retry if value is None (etcd consistency delay)
+            start_time = time.time()
+            while val is None:
+                if time.time() - start_time > timeout_sec:
+                    raise TimeoutError(
+                        f"[Rank {self.rank}] alltoall_obj timeout waiting for rank {src_rank} after {timeout_sec}s"
+                    )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)